extensions/HyperVExtension/src/HyperVExtension/Services/DownloaderService.cs

```diff
@@ -71,4 +130,17 @@ private long GetTotalBytesToReceive(HttpResponseMessage response)
         // We should be able to get the content length from the response headers from the Microsoft servers.
         throw new InvalidOperationException("The content length of the response is not known.");
     }
+
+    public bool IsFileBeingDownloaded(string destinationFilePath)
+    {
+        lock (_lock)
+        {
+            if (_destinationFileDownloadMap.TryGetValue(destinationFilePath, out var subscribers))
+            {
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
In StartDownloadAsync this lock is taken for the duration of the downloading process, so this method won't return until downloading is done.

Reply:
From what I see I don't see how that would be the case. I don't lock the entire StartDownloadAsync method. From my testing, this doesn't get locked for the entirety of the StartDownloadAsync  call. But please let me know if I'm missing something. Thanks